### ISSUE TITLE
Support heredoc syntax

### ISSUE THREAD

**Issue by AlanFoster** [iamsolankiamit/prettier-ruby#41]
## Input

```ruby
def sql
  <<-SQL
    SELECT *
    FROM foo
  SQL
end
```

## Current Output

```ruby
def sql
  '    SELECT *
    FROM foo
'
end
```

## Expected Output

The same as the current input. In the future perhaps `prettier-ruby` could apply SQL formatting to SQL heredocs 🤷‍♀️ 

## Tokens

```ruby
       [[[1, 0], :on_kw, "def"],
       [[1, 3], :on_sp, " "],
       [[1, 4], :on_ident, "sql"],
       [[1, 7], :on_nl, "\n"],
       [[2, 0], :on_sp, "  "],
       [[2, 2], :on_heredoc_beg, "<<-SQL"],
       [[2, 8], :on_nl, "\n"],
       [[3, 0], :on_tstring_content, "    SELECT *\n    FROM foo\n"],
       [[5, 0], :on_heredoc_end, "  SQL\n"],
       [[6, 0], :on_kw, "end"],
       [[6, 3], :on_nl, "\n"]]
```

**Comment by j-f1** [comment]
> In the future perhaps prettier-ruby could apply SQL formatting to SQL heredocs 🤷‍♀️

Prettier core already does this for tagged template strings in Markdown — perhaps you could look at that implementation to figure out how to do this.

**Comment by AlanFoster** [comment]
@j-f1 I had my eye on the `embed` function in Prettier's API when writing this issue:

https://github.com/prettier/prettier/blob/70604f1fd172b599378982b91216b94e738144ab/docs/plugins.md#printers 

It would be nice to add, but definitely doesn't need to be in the scope of this issue haha, just wishful thinking for now

**Comment by iamsolankiamit** [comment]
Fixed in 0d9312c

Right now we output the same way. we can look into `embed` function in the future. can you guys (@j-f1 , @AlanFoster ) create a sperate issue regarding that?
